src/pages/EarnTri/index.tsx

```diff
@@ -1,143 +1,198 @@
-  display: grid;
-  grid-template-columns: 1fr 1fr;
-  gap: 20px;
-  width: 100%;
-  justify-self: center;
-  
-  ${({ theme }) => theme.mediaWidth.upToSmall`
-    grid-template-columns: 1fr;
- `};
-`
-
-const DataRow = styled(RowBetween)`
-  ${({ theme }) => theme.mediaWidth.upToSmall`
-   flex-direction: column;
- `};
-`
-
-const SortSection = styled.div`
-  display: flex;
-`
-const SortField = styled.div`
-  margin: 0px 5px 0px 5px;
-  font-weight: 400;
-  font-size: 14px;
-  cursor: pointer;
-  display: flex;
-  line-height: 20px;
-`
-
-const SortFieldContainer = styled.div`
-  display: flex;
-  ${({ theme }) => theme.mediaWidth.upToSmall`
-   display: none;
- `};
-`
+import { TYPE } from '../../theme'
+import { isTokenAmountPositive } from '../../utils/pools'
+
+import {
+  PoolSection,
+  DataRow,
+  StyledSearchInput,
+  StyledFiltersContainer,
+  StyledToggleContainer,
+  StyledSortContainer,
+  StyledSortOption,
+  StyledArrowContainer
+} from './EarnTri.styles'
 
 enum SortingType {
-  totalStakedInWavax = 'totalStakedInWavax',
-  multiplier = 'multiplier',
-  totalApr = 'totalApr'
+  liquidity = 'Liquidity',
+  totalApr = 'Total APR',
+  default = 'Default'
 }
 
-const POOLS_ORDER = [5, 11, 8, 7, 0, 1, 2, 3, 4, 9, 10, 12, 13, 14];
+const POOLS_ORDER = [5, 11, 8, 7, 0, 1, 2, 3, 4, 9, 10, 12, 13, 14]
 const LEGACY_POOLS = [6]
 
+const MemoizedFarmBanner = React.memo(FarmBanner)
+const MemoizedPoolCardTRI = React.memo(PoolCardTRI)
+
 export default function Earn({
   match: {
     params: { version }
   }
 }: RouteComponentProps<{ version: string }>) {
   const { t } = useTranslation()
-  const farmArrs = useFarms();
-  const [searchQuery, setSearchQuery] = useState<string>('');
-  const farmArrsInOrder = POOLS_ORDER.map(index => farmArrs[index]);
-  const legacyFarmArrsInOrder = LEGACY_POOLS.map(index => farmArrs[index]);
+  const allFarmArrs = useFarms()
+  const toggleActiveFarms = useToggleFilterActiveFarms()
+  const activeFarmsFilter = useIsFilterActiveFarms()
 
-  const dualRewardPools = farmArrsInOrder.filter(farm => farm.doubleRewards)
+  const [sortBy, setSortBy] = useState<SortingType>(SortingType.default)
+  const [searchQuery, setSearchQuery] = useState<string>('')
+  const [sortDescending, setSortDescending] = useState<boolean>(true)
+
+  const getSortedFarms = () => {
+    switch (sortBy) {
+      case SortingType.default:
+        return POOLS_ORDER.map(index => allFarmArrs[index])
+    }
+    switch (sortBy) {
+      case SortingType.liquidity:
+        return sortDescending
+          ? farmArrs.sort((a, b) => (a.totalStakedInUSD < b.totalStakedInUSD ? 1 : -1))
+          : farmArrs.sort((a, b) => (a.totalStakedInUSD > b.totalStakedInUSD ? 1 : -1))
+    }
+    switch (sortBy) {
+      case SortingType.totalApr:
+        return sortDescending
+          ? farmArrs.sort((a, b) => (a.apr + a.apr2 < b.apr + b.apr2 ? 1 : -1))
+          : farmArrs.sort((a, b) => (a.apr + a.apr2 > b.apr + b.apr2 ? 1 : -1))
+    }
```

Comment:
Was there a reason for multiple `switch`s?

```suggestion
    switch (sortBy) {
      case SortingType.default:
        return POOLS_ORDER.map(index => allFarmArrs[index])
      case SortingType.liquidity:
        return sortDescending
          ? farmArrs.sort((a, b) => (a.totalStakedInUSD < b.totalStakedInUSD ? 1 : -1))
          : farmArrs.sort((a, b) => (a.totalStakedInUSD > b.totalStakedInUSD ? 1 : -1))
      case SortingType.totalApr:
        return sortDescending
          ? farmArrs.sort((a, b) => (a.apr + a.apr2 < b.apr + b.apr2 ? 1 : -1))
          : farmArrs.sort((a, b) => (a.apr + a.apr2 > b.apr + b.apr2 ? 1 : -1))
    }
```

Reply:
messed up!